File: LongWords.cpp
#include <bits/stdc++.h>
using namespace std;

int main(void) {
    int wordCount;
    cin >> wordCount;
    string word;
    vector<string> wordsArray;
    for (int i=0; i<wordCount; i++) {
        cin >> word;
        if (word.size() > 10) {
            string tempWord;
            int temp = word.size() - 2;
            string wordLength = std::to_string(temp);
            tempWord.push_back(word[0]);
            tempWord += wordLength;
            tempWord.push_back(word.back());
            wordsArray.push_back(tempWord);
        }
        else  
            wordsArray.push_back(word);
    }
    for (auto i=wordsArray.begin(); i!=wordsArray.end(); i++)
        cout << *i << endl;
    
    return 0;
}

File: Python/LongWords.py
def main():
    wordCount = int(input())
    word = ""
    wordsArray = []
    for i in range(wordCount):
        word = input()
        if len(word) > 10:
            tempWord = ""
            temp = len(word)-2
            wordLength = str(temp)
            tempWord += word[0]
            tempWord += str(wordLength)
            tempWord += word[-1]
            wordsArray.append(tempWord)
        else:
            wordsArray.append(word)
    for i in wordsArray:
        print(i)


if __name__ == '__main__':
    main()
